opm/autodiff/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -195,7 +195,9 @@ namespace Opm
             data::Wells res = WellState::report();
 
             const int nw = this->numWells();
-            const int np = this->numPhases();
+            // If there are now wells numPhases throws a floating point
+            // exception.
+            const int np = nw ? this->numPhases() : -1;
```

Comment:
I would prefer to leave at as is. Only one if branch and one division
in this case.